File: FacturApp/FacturApp/FacturApp/Models/Cliente.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FacturApp.Models
{
    public class Cliente
    {
        public string nombre { get; set; }
        public string dnicif { get; set; }
        public string direccion { get; set; }
        public int telefono { get; set; }
        public string email { get; set; }

        public Cliente()
        {
            nombre = "";
            dnicif = "";
            direccion = "";
            telefono = 111111111;
            email = "";
        }

        public Cliente(string nombre, string dnicif, string direccion, int telefono, string email)
        {
            this.nombre = nombre;
            this.dnicif = dnicif;
            this.direccion = direccion;
            this.telefono = telefono;
            this.email = email;
        }

        public Cliente(Cliente cliente)
        {
            this.nombre = cliente.nombre;
            this.dnicif = cliente.dnicif;
            this.direccion = cliente.direccion;
            this.telefono = cliente.telefono;
            this.email = cliente.email;
        }
    }
}

File: FacturApp/app/src/main/java/com/gcs/facturapp/models/Cliente.java
package com.gcs.facturapp.models;

import java.io.Serializable;

public class Cliente implements Serializable{
    public String nombre;
    public String apellidos;
    public String dnicif;
    public String direccion;
    public String telefono;
    public String email;

    public Cliente(){}

    public Cliente(String nombre, String apellidos, String dnicif, String direccion, String telefono, String email)
    {
        this.nombre = nombre;
        this.apellidos = apellidos;
        this.dnicif = dnicif;
        this.direccion = direccion;
        this.telefono = telefono;
        this.email = email;
    }

    public Cliente(Cliente cliente)
    {
        this.nombre = cliente.nombre;
        this.apellidos = cliente.apellidos;
        this.dnicif = cliente.dnicif;
        this.direccion = cliente.direccion;
        this.telefono = cliente.telefono;
        this.email = cliente.email;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (this == obj) return true;
        if (obj instanceof Cliente) {
            Cliente cli = (Cliente)obj;
            return this.nombre.equals(cli.nombre) && this.apellidos.equals(cli.apellidos) || this.dnicif.equals(cli.dnicif);
        }

        return false;
    }

    @Override
    public String toString()
    {
        return this.nombre + " " + this.apellidos;
    }
}
